libs/secrets/CMakeLists.txt

```diff
@@ -36,7 +38,7 @@ if (USE_JSON AND USE_HTTPLIB)
                                                 PRIVATE src
     )
     target_compile_definitions(secret_retriever PUBLIC $<$<BOOL:${HTTPLIB_IS_USING_OPENSSL}>:CPPHTTPLIB_OPENSSL_SUPPORT>)
-    target_link_libraries(secret_retriever PRIVATE nlohmann_json::nlohmann_json ${OPENSSL_LIBRARIES})
+    target_link_libraries(secret_retriever PUBLIC nlohmann_json::nlohmann_json OpenSSL::SSL concord-crypto)
```

Comment:
Didn't you say that you needed only to link against concord-crypto to get openssl?

Reply:
Correct, therefore the "almost" in the commit message. `concord-crypto` currently brings only OpenSSL::Crypto library